models/git_diff.go

```diff
@@ -164,11 +163,11 @@ func ParsePatch(pid int64, maxlines int, cmd *exec.Cmd, reader io.Reader) (*Diff
 		// Get new file.
 		if strings.HasPrefix(line, DIFF_HEAD) {
 			if isTooLong {
-				return diff, nil
+				break
 			}
 
-			fs := strings.Split(line[len(DIFF_HEAD):], " ")
-			a := fs[0]
+			beg := len(DIFF_HEAD)
+			a := line[beg : (len(line)-beg)/2+beg]
```

Comment:
Yes, because these two are the same file with different prefixes like `a/file` and `b/file`. A string split on " b/" is the stronger alternative (than current " ") but could fail on some file names too.

Someone else has to confirm if file rename is impl with remove + add by git though.